src/main/java/backend/Logic.java

```diff
@@ -206,14 +201,41 @@ public Model getRepo(String repoId) {
      * @return True if label replacement on GitHub was a success, false otherwise.
      */
     public CompletableFuture<Boolean> replaceIssueLabels(TurboIssue issue, List<String> newLabels) {
+        boolean isLabelsChanged = updateIssueLabels(issue, newLabels);
+
+        if (isLabelsChanged) {
+            logger.info("Changing labels for " + issue + " on UI");
+            refreshUI();
+        }
+
+        return updateIssueLabelsOnRepo(issue, newLabels);
+    }
```

Comment:
Ok sure. I will refactor these. Also, this "almost correct" behaviour doesn't feel so good so I will try to deal with the clash with periodic update and put in some fallback measures, but likely won't make it for this milestone though.